### ISSUE TITLE
Target extras don't exist when target_post_save hook runs on target creation

### ISSUE THREAD

**Issue by jfrostburke** [TOMToolkit/tom_base#202]
I'm having the problem again where target extras (which now includes `target.names`) don't exist when the target gets created, so the `target_post_save` hook can't access them.

In my case specifically, I updated the `target_post_save` hook to check if any of the `target.names` start with `Gaia` so that I can import Gaia photometry, as suggested by Łukasz. When I use my TOM's web interface, if I add a Gaia name as an alias on target creation, the photometry does not get imported. If I click the "Update Target" button, change nothing, and submit, then it gets imported and displayed in my lightcurve plot correctly. 

If the target extras were already associated with the target when the `target_post_save` hook runs on target creation, this would make my photometry importing smoother, and would also allow me to finally make my Twitterbot.

Specific code I used to check for the Gaia name in case it is useful:
```python
  gaia_name = next((name for name in target.names if 'Gaia' in name), None)
  if gaia_name:
    do stuff
```

**Comment by jfrostburke** [comment]
I still don't have a solution, but updates after some digging:

In `tom_targets.Target.save`, there's a for-loop running over names and saving them. However,  names only get saved there if you are using your TOM programmatically, i.e. with `target.save(names=[name1,name2,...])`. If you're using the TOM through the web interface, then the names are accessed through `tom_targets.forms.TargetNamesFormset`, and they get saved in a separate place, in `tom_targets.views.Target{Create,Update}View.form_valid`.

I've imported `run_hook` into `views.py` and confirmed that putting it after the names and extras get saved results in the expected behavior, i.e. using `target.names` inside the hook returns the full set of target names. But this is a pretty janky solution, since the hook runs twice: once with an incomplete name list when the target gets saved, and once correctly when the form gets validated.

Anyways, still not sure what the right thing to do is but at least I understand the code a little bit better.

**Comment by dmcollom** [comment]
I was looking into this a little earlier this week. I have an active branch that fixes the TargetUpdateView by moving the `super().save()` in the `form_valid` below the validation/saving of the extras/names and decorating the form with `@transaction.atomic`, and that doesn't appear to cause any other issues with the validation of names and extras.

It's not immediately clear to me how this should be solved for the TargetCreateView, though--obviously the logic for validating/saving extras/names can't be moved up without an existing foreign key.

**Comment by dmcollom** [comment]
This has been addressed by PR #209 and has been released in 1.2.0.
